tools/image-builder/templates/packer_user_data.tpl

```diff
@@ -0,0 +1,151 @@
+    done
+
+    if [ $count -eq "$retries" ]; then
+        echo "Failed to install after $retries attempts. Exiting."
+        exit 1
+    fi
+}
+
+echo $'***** Installing s3fs *****\n'
+sudo rpm --import https://dl.fedoraproject.org/pub/epel/RPM-GPG-KEY-EPEL-8
+sudo dnf install -y https://dl.fedoraproject.org/pub/epel/epel-release-latest-8.noarch.rpm
+install_with_retry "sudo yum install -y s3fs-fuse" 3
+rpm -qa | grep epel-release
+rpm -qa | grep s3fs-fuse
+
+echo "======================Cloning HPC public repo====================="
+
+sudo yum install git -y
+mkdir /HPCaaS
+cd /HPCaaS
+git clone https://github.com/terraform-ibm-modules/terraform-ibm-hpc.git
+cd /HPCaaS/terraform-ibm-hpc/solutions/hpc
+
+echo "======================Installing terraform====================="
+git clone --depth=1 https://github.com/tfutils/tfenv.git ~/.tfenv
+echo "export PATH=$PATH:$HOME/.tfenv/bin" >> ~/.bashrc
+ln -s ~/.tfenv/bin/* /usr/local/bin
+tfenv install 1.5.7
+tfenv use 1.5.7
+terraform --version
+
+echo "====================== Triggering mounting of Cos Bucket ====================="
+mkdir /wes-hpc
+s3fs custom-image-builder /wes-hpc -o url=https://s3.direct.us-south.cloud-object-storage.appdomain.cloud -o ro -o public_bucket=1
+mkdir -p /HPCaaS/terraform-ibm-hpc/tools/tests
+cp -r /wes-hpc/tests/* /HPCaaS/terraform-ibm-hpc/tools/tests/
+ls -ltr /HPCaaS/terraform-ibm-hpc/tools/tests/
+echo "====================== Cos Bucket mounting completed ====================="
+
+cd /var/packer/hpcaas/compute
+
+sudo -E packer init . && sudo -E packer build \
+    -var "ibm_api_key=${ibm_api_key}" \
+    -var "vpc_region=${vpc_region}" \
+    -var "resource_group_id=${resource_group_id}" \
+    -var "vpc_subnet_id=${vpc_subnet_id}" \
+    -var "source_image_name=${source_image_name}" \
+    -var "install_sysdig=${install_sysdig}" \
+    -var "security_group_id=${security_group_id}" \
+    -var "image_name=${image_name}" .
+
+echo "========== Generating SSH key ========="
+mkdir -p /HPCaaS/artifacts/.ssh
+ssh-keygen -t rsa -N '' -f /HPCaaS/artifacts/.ssh/id_rsa <<< y
+
+RANDOM_SUFFIX=$(head /dev/urandom | tr -dc 'a-z' | head -c 4)
+CICD_SSH_KEY="hpc-packer-$RANDOM_SUFFIX"
+
+PACKER_FIP=$(curl -s ifconfig.io)
+
+echo "========== Installing IBM cloud CLI ========="
+curl -fsSL https://clis.cloud.ibm.com/install/linux | sh
+ibmcloud plugin install infrastructure-service
+ibmcloud login --apikey ${ibm_api_key} -r ${vpc_region}
+echo "========== Uploading SSH key to IBM cloud ========="
+ibmcloud is key-create $CICD_SSH_KEY @/HPCaaS/artifacts/.ssh/id_rsa.pub --resource-group-name ${resource_group}
+
+cd /HPCaaS/terraform-ibm-hpc/tools/tests
+git submodule update --init
+
+sudo yum update -y
+
+# Download the installer script:
+curl --proto '=https' --tlsv1.2 -fsSL https://get.opentofu.org/install-opentofu.sh -o install-opentofu.sh
+
+# Give it execution permissions:
+chmod +x install-opentofu.sh
+
+# Run the installer:
+./install-opentofu.sh --install-method rpm
```

Comment:
Looks like you are installing Opentofu in order to use it with modules from TIM. This will not work, it might work today but we are already seeing the two versions diverge and I expect within the next year or two there will be incompatibilities. For example, it is possible to write OpenTofu configuration that is not compatible with Terraform today.

Reply:
We actually needed opentofu to run the testcases. If any alternative way is suggested let that know.

Reply:
This seems to be a bug in the testwrapper causing the requirement. Updating the wrapper to see if it fixes the requirement.

Reply:
Yes, I expect it's a bug, hopefully, it is resolved in a later version. We have not seen the bug before so I am hoping it got resolved as a side effect of other refactoring work. That version is over a month old and no reports of this bug.